src/renderer/components/resizing-anchor/resizing-anchor.tsx

```diff
@@ -0,0 +1,280 @@
+  onDrag?: (newExtent: number) => void;
+
+  // onDoubleClick is called when the the ResizeAnchor is double clicked
+  onDoubleClick?: () => void;
+
+  /**
+   * The following two extents represent the max and min values set to `onDrag`
+   */
+  maxExtent?: number;
+  minExtent?: number;
+
+  /**
+   * The following events are triggerred with respect to the above values.
+   *  - The "__Exceed" call will be made when the unbounded extent goes from
+   *    < the above to >= the above
+   *  - The "__Subceed" call is similar but is triggered when the unbounded
+   *    extent goes from >= the above to < the above.
+   */
+  onMaxExtentExceed?: () => void;
+  onMaxExtentSubceed?: () => void;
+  onMinExtentSubceed?: () => void;
+  onMinExtentExceed?: () => void;
+}
+
+enum MovementTrigger {
+  EXCEED_MAX = "exceed",
+  SUBCEED_MIN = "subceed",
+}
+
+interface MovementCalc {
+  newExtent?: number;
+  otherTrigger?: MovementTrigger;
+  ignore?: boolean;
+}
+
+interface Position {
+  readonly pageX: number;
+  readonly pageY: number;
+}
+
+/**
+ * Return the direction delta, but ignore drags leading up to a moved item
+ *  1. `->|` => return `false`
+ *  2. `<-|` => return `directed length (P1, P2)` (negative)
+ *  3. `-|>` => return `directed length (M, P2)` (positive)
+ *  4. `<|-` => return `directed length (M, P2)` (negative)
+ *  5. `|->` => return `directed length (P1, P2)` (positive)
+ *  6. `|<-` => return `false`
+ * @param P the starting position on the number line
+ * @param Q the ending position on the number line
+ * @param M a third point that determines if the delta is meaningful
+ * @returns the directional difference between including appropriate sign.
+ */
+function directionDelta(P1: number, P2: number, B: number): number | false {
+  if (P1 < B) {
+    if (P2 >= B) {
+      // case 3
+      return Math.abs(B - P2)
+    }
+
+    if (P2 < P1) {
+      // case 2
+      return -Math.abs(P1 - P2)
+    }
+
+    // case 1
+    return false
+  }
+
+  if (P2 < B) {
+    // case 4
+    return -Math.abs(B - P2)
+  }
+
+  if (P1 < P2) {
+    // case 5
+    return Math.abs(P1 - P2)
+  }
+
+  // case 6
+  return false
+}
+
+export class ResizingAnchor extends React.PureComponent<Props> {
+  @observable lastMouseEvent?: MouseEvent
+  @observable elem: HTMLElement
+
+  static defaultProps = {
+    onStart: noop,
+    onDrag: noop,
+    onEnd: noop,
+    onMaxExtentExceed: noop,
+    onMinExtentExceed: noop,
+    onMinExtentSubceed: noop,
+    onMaxExtentSubceed: noop,
+    onDoubleClick: noop,
+    disabled: false,
+    growthDirection: ResizeGrowthDirection.BOTTOM_TO_TOP,
+    maxExtent: Number.POSITIVE_INFINITY,
+    minExtent: 0,
+    placement: ResizeSide.LEADING,
+  }
+  static IS_RESIZING = "resizing"
+
+  constructor(props: Props) {
+    super(props)
+    if (props.maxExtent < props.minExtent) {
+      throw new Error("maxExtent must be >= minExtent")
+    }
+  }
+
+  componentDidMount() {
+    this.elem = findDOMNode(this) as HTMLElement
```

Comment:
Maybe we should use `ref` instead since it's recommended not to use `findDOMNode` in most cases
https://reactjs.org/docs/react-dom.html#finddomnode
https://reactjs.org/docs/refs-and-the-dom.html#gatsby-focus-wrapper

Reply:
Thanks for the pointers